components/blocks/SiteFooter/LanguageSelector/LanguageSelector.test.ts

```diff
@@ -0,0 +1,21 @@
+import { type LocaleObject } from '@nuxtjs/i18n/dist/runtime/composables'
+import { mount } from '@vue/test-utils'
+import { mockI18n } from 'root/vitest.setup'
+import LanguageSelector from './LanguageSelector.vue'
+
+const mockedI18n = mockI18n()
```

Comment:
CMIIW, but can we call `useI18n` here, since we stub `useI18n` with `mockI18n` in `vitest.setup.ts`, or is that not how it works?

Reply:
@zysim You're right, we could have done that and it would've known the right thing to use. Interestingly, in doing that the returned type of the `locales` object was a `ComputedRef<LocaleObject[]>`, which was slightly problematic. Typically, you should have to call `value` on a computed reference in order to get the actual value out, but in this case that was returning `undefined` because of how we stubbed it out for the tests. 

After spending about 20 minutes trying to figure a work around to satisfy TypeScript without needing to cast the `locales` object as `unknown`. Ultimately, I realized a simpler and more explicit test works better here, since we're trying to make sure that we're both stubbing the `useI18n` module correctly and that component renders the data correctly. Doing it the way seen in 1510b5d283a60e06fd9f4c904b5504f0b404bc81 accomplishes this, as we now know that the stub itself and the component rendering are correct (since before it could've been returning an empty array, which we were then using to check to see if it rendered correctly, effectively not testing anything).

Reply:
Ah okay. Thanks for going through the whole trouble of figuring this out! Appreciate the effort